projects/ngx-maplibre-gl/src/lib/source/geojson/feature.component.ts

```diff
@@ -1,53 +1,55 @@
 import {
   Component,
   forwardRef,
-  Inject,
-  Input,
   OnDestroy,
   OnInit,
   ChangeDetectionStrategy,
+  input,
+  model,
+  inject,
 } from '@angular/core';
 import { GeoJSONSourceComponent } from './geojson-source.component';
 
 /**
  * `mgl-feature` - a feature component
  * [ngx] inside {@link GeoJSONSourceComponent} only
- * 
+ *
  * @category Source Components
  */
 @Component({
-    selector: 'mgl-feature',
-    template: '',
-    changeDetection: ChangeDetectionStrategy.OnPush,
-    standalone: true,
+  selector: 'mgl-feature',
+  template: '',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
 })
-export class FeatureComponent
-  implements OnInit, OnDestroy, GeoJSON.Feature<GeoJSON.GeometryObject> {
+export class FeatureComponent implements OnInit, OnDestroy {
+  /** Init injection */
+  private readonly geoJSONSourceComponent = inject<GeoJSONSourceComponent>(
+    forwardRef(() => GeoJSONSourceComponent)
+  );
   /** Init input */
-  @Input() id?: number; // FIXME number only for now https://github.com/mapbox/mapbox-gl-js/issues/2716
+  readonly id = model<number>(); // FIXME number only for now https://github.com/mapbox/mapbox-gl-js/issues/2716
```

Comment:
I think this comment should be removed, I don't think someone will fix this in the near future...